trl/trainer/sft_trainer.py

```diff
@@ -500,16 +514,6 @@ def _func(example):
                     )
                     dataset = dataset.map(_func, batched=True, **map_kwargs)
 
-            # If the dataset is prompt-completion, convert it to language modeling type
-            first_example = next(iter(dataset))
-            if "prompt" in first_example.keys() and "completion" in first_example.keys():
-                key = "messages" if is_conversational(first_example) else "text"
-
-                def concat_prompt_completion(example):
-                    return {key: example["prompt"] + example["completion"]}
-
-                dataset = dataset.map(concat_prompt_completion, remove_columns=["prompt", "completion"])
-
```

Comment:
This concatenation needs to be removed, as we loses the information about where the completion starts. This completion is now managed in `tokenize`.